
### Primers and PCR Conditions 
The tables below show primers and PCR conditions for **Pfcrt (6.1), Pfk13 (6.2), Mitochondrial genome (6.3), Pfcytb (6.3a), Pfdhps (6.4), Pfdhfr (6.5), Pfmdr1 (6.6).**

**Table 6.1 *Pfcrt***

|*Pfcrt (3109bp)*    | **50.0 μlrxn**| X samples+1 |	***Final [conc]***  |                                           
|:------------------:|:--------------:|:-------------:|:--------------:|
| *Master Mix:*      |                                                                                                 	                                                             
| *5X GC Buffer*    | 	*10.0 μl*|                  | *1 x*|
| *dNTPs (10mM)*	  |   *1 μl*	 |                  | *0.2 mM*|
| *F primer (10uM)* |	*1.25 μl*	 |                  | *0.25 uM*|
| *R primer (10uM)*	|  *1.25 μl* |                  |*0.25 uM*|
 |*H<sub>2</sub>O*	            |  *32 μl*   |                  |         |
 |**Add last:** 	  |            |                  |         |
 |*HF Phusion Taq*  |	  *0.5μl*  |                  |*1 unit*|
 |*Total*	          | **46.0μl** |                  |           |
 |**Add:**		      |            |                   ||
| *Template DNA*	  |*4.0 μl/well*| 	 ------	      |*50-250 ng*|
 |**TOTAL**	          |**50.0 μl**	 |	                 |          |


|Thermal  |cycling  |conditions|
|:------:|:---------:|:----------:|
|*98 oC*        |*3:00 min*  |           |
|***98 oC***	|**_0:10 min_**|           |
|***62 oC***	|**_0:30 min_**| **_30X_** |
|***65 oC***	|**_5:00 min_**|       |
|*65 oC*	|*10:00min*	 |           |
|*4 oC* |*∞*|        |   




**Primers:  Pfcrt_F1 Sequence (5' to 3'):** TTACATATAACAAAATGAAATTCGC; **Pfcrt_R1 Sequence (5' to 3'):** TATTGTGTAATAATTGAATCGACG


**Table 6.2 _Pfk13_**

|*PfK13 (2120bp)*	  | **50.0 μlrxn**| X samples+1 |	***Final [conc]***  |                                           
|:------------------:|:--------------:|:-------------:|:--------------:|
| *Master Mix:*      |                                                                                                 	                                                              | *5X GC Buffer*     | 	*10.0 μl*|                  | *1 x*|
| *dNTPs (10mM)*	  |   *1 μl*	 |                  | *0.2 mM*|
| *F primer (10uM)*  |	*1.25 μl*	 |                  | *0.25 uM*|
| *R primer (10uM)*	|  *1.25 μl* |                  |*0.25 uM*|
 |*H<sub>2</sub>O*	 |  *32 μl*   |                  |         |
 |**Add last:** 	  |            |                  |         |
 |*HF Phusion Taq*  |	  *0.5μl*  |                  |*1 unit*|
 |*Total*	          | **46.0μl** |                  |           |
 |**Add:**		      |            |                   ||
| *Template DNA*	  |*4.0 μl/well*| 	 ------	      |*50-250 ng*|
 |**TOTAL**	          |**50.0 μl**	 |	                 |          |


|Thermal  |cycling  |conditions|
|:------|:---------|:----------|
|*98 oC*	  |*3:00 min*	   |           |
|***98 oC***	|***0:10 min***|           |
|***58 oC***	|***0:30 min***	 | ***30X*** |
|***65 oC***	|***5:00 min***	 |       |
|*65 oC*	|*10:00min*	 |           |
|*4 oC* |*∞*|        |   



**Primers:  Pfk13_F1 Sequence (5' to 3'):** CTATGACGTATGATAGGGAATCTGG; **Pfk13_R1 Sequence (5' to 3'):** CTGGGAACTAATAAAGATGGGCC




**Table6.3 Mitochondrial genome**

|*Mitch(5967bp)*	  | **50.0 μlrxn**| X samples+1 |	***Final [conc]***  |                                           
|:--------------------:|:--------------:|:-------------:|:-----------------:|
| *Master Mix:*      |                                                                                                 	                                               
| *5X GC Buffer*    | 	*10.0 μl*|                  | *1 x*|
| *dNTPs (10mM)*	  |   *1 μl*	 |                  | *0.2 mM*|
| *F primer (10uM)* |	*1.25 μl*	 |                  | *0.25 uM*|
| *R primer (10uM)*	|  *1.25 μl* |                  |*0.25 uM*|
 |*H<sub>2</sub>O*	            |  *32 μl*   |                  |         |
 |**Add last:** 	  |            |                  |         |
 |*HF Phusion Taq*  |	  *0.5μl*  |                  |*1 unit*|
 |*Total*	          | **46.0μl** |                  |           |
 |**Add:**		      |            |                   ||
| *Template DNA*	  |*4.0 μl/well*| 	 ------	      |*50-250 ng*|
 |**TOTAL**	          |**50.0 μl**	 |	                 |          |
 
 |Thermal  |cycling  |conditions|
|:------|:---------|:----------|
|*98 oC*	  |*3:00 min*	   |           |
|***98 oC***	|***0:10 min***|           |
|***50 oC***	|***0:30 min***	 | ***30X*** |
|***65 oC***	|***5:00 min***	 |       |
|*65 oC*	|*10:00min*	 |           |
|*4 oC* |*∞*|        |   




**Primers: Mitochondrion_F1 Sequence (5’ to 3’):** AAGCTTTTGGTATCTCGTAAT; **Mitochondrion_R1 Sequence (5’ to 3’):** TATTATAATATAACTCTACAAAGTTGAAC




**NOTE:** If experiencing issues with amplifying the full-length mitochondrial genome, consider amplifying only the cyt-b gene instead for characterizing molecular markers associated with Malarone (atovaquone/proguanil) resistance. See Table 6.3a below. 




**Table6.3a *Cytochrome b***


|*Pfcytb 1 (937bp)*	 | **50.0 μlrxn**| X samples+1 |	***Final [conc]***  |                                           
|:------------------:|:--------------:|:-------------:|:--------------:|
| *Master Mix:*      |                                                                                                 	                                                       
| *5X GC Buffer*     | 	*10.0 μl*      |                  | *1 x*|
| *dNTPs (10mM)*	   |   *1 μl*	     |                  | *0.2 mM*|
| *F primer (10uM)*  |	*1.25 μl*	  |                  | *0.25 uM*|
| *R primer (10uM)*	 |  *1.25 μl*  |                  |*0.25 uM*|
| *H<sub>2</sub>O*    |  *32 μl*   |                  |         |
 |**Add last:** 	   |             |                  |         |
 |*HF Phusion Taq*   |	  *0.5μl*  |                  |*1 unit*|
 |*Total*	           | **46.0μl**  |                  |           |
 |**Add:**		       |             |                   ||
 | *Template DNA*	   |*4.0 μl/well*| 	 ------	      |*50-250 ng*|
 |**TOTAL**	          |**50.0 μl**	 |	                 |          |

|Thermal  |cycling  |conditions|
|:------|:---------|:----------|
|*98 oC*	  |*3:00 min*	   |           |
|***98 oC***	|***0:10 min***|           |
|***60 oC***	|***0:30 min***	 | ***30X*** |
|***65 oC***	|***5:00 min***	 |       |
|*65 oC*	|*10:00min*	 |           |
|*4 oC* |*∞*|        |   


**Primers: CytB_F1_Sequence (5' to 3'):** CTATTAATTTAGTTAAAGCACAC; **CytB_R1_Sequence (5' to 3'):** ACAGAATAATCTCTAGCACCA


**Table 6.4 *mdr1***

|*Pfmdr1 (4155bp)*	  | **50.0 μlrxn**| X samples+1 |	***Final [conc]***  |                                           
|:------------------|:--------------:|:-------------|:--------------:|
| *Master Mix:*      |                                                                                                 	                                               
| *5X GC Buffer*    | 	*10.0 μl*|                  | *1 x*|
| *dNTPs (10mM)*	  |   *1 μl*	 |                  | *0.2 mM*|
| *F primer (10uM)* |	*1.25 μl*	 |                  | *0.25 uM*|
| *R primer (10uM)*	|  *1.25 μl* |                  |*0.25 uM*|
 |*H2O*	            |  *32 μl*   |                  |         |
 |**Add last:** 	  |            |                  |         |
 |*HF Phusion Taq*  |	  *0.5μl*  |                  |*1 unit*|
 |*Total*	          | **46.0μl** |                  |           |
 |**Add:**		      |            |                   ||
| *Template DNA*	  |*4.0 μl/well*| 	 ------	      |*50-250 ng*|
 |**TOTAL**	          |**50.0 μl**	 |	                 |          |

|Thermal  |cycling  |conditions|
|:----------:|:---------:|:----------:|
|*98 oC*	  |*3:00 min*	   |           |
|***98 oC***	|***0:10 min***|           |
|***60 oC***	|***0:30 min***	 | ***30X*** |
|***65 oC***	|***5:00 min***	 |       |
|*65 oC*	|*10:00min*	 |           |
|*4 oC* |*∞*|        |   


**Primers:  Pfmdr1_F1_Sequence (5' to 3'):** TGGTAACCTCAGTATCAAAG; **Pfmdr1_R1_Sequence** (5' to 3'): CATCTTGTGCTGATAATAATTC





**Table 6.5 *dhfr***

|*Pfdhfr (2067bp)*| **50.0 μlrxn**| X samples+1 |	***Final [conc]***  |                                          
|:------------------:|:--------------:|:-------------:|:--------------:|
| *Master Mix:*      |                                                                                                 	                                      
| *5X GC Buffer*    | 	*10.0 μl*|                  | *1 x*|
| *dNTPs (10mM)*	  |   *1 μl*	 |                  | *0.2 mM*|
| *F primer (10uM)* |	*1.25 μl*	 |                  | *0.25 uM*|
| *R primer (10uM)*	|  *1.25 μl* |                  |*0.25 uM*|
 |*H<sub>2</sub>O*	            |  *32 μl*   |                  |         |
 |**Add last:** 	  |            |                  |         |
 |*HF Phusion Taq*  |	  *0.5μl*  |                  |*1 unit*|
 |*Total*	          | **46.0μl**	 |                  |           |
 |**Add:**		      |            |                   ||
| *Template DNA*	  |*4.0 μl/well*| 	 ------	      |*50-250 ng*|
 |**TOTAL**	        |**50.0 μl**	|	                 |          |

|Thermal  |cycling  |conditions|
|:------:|:---------:|:----------:|
|*98 oC*	  |*3:00 min*	   |           |
|***98 oC***	|***0:10 min***|           |
|***58 oC***	|***0:30 min***	 | ***30X*** |
|***65 oC***	|***5:00 min***	 |       |
|*65 oC*	|*10:00min*	 |           |
|*4 oC* |*∞*|        |   

**Primers:  Pfdhfr_F1 Sequence (5' to 3'):** TTTTTACTAGCCATTTTTGTATTCC; **Pfdhfr_R1 Sequence  (5' to 3'):** TTAACCGTTCAGGTAATTTTGTCA

***\*Primers adapted from: SC, Carlton JM. 2016.*** A Method for Amplicon Deep Sequencing of Drug Resistance Genes in Plasmodium falciparum Clinical Isolates from India. J Clin Microbiol 54:1500–1511.





**Table 6.6 *dhps***

|*Pfdhps (2817bp)*	  | **50.0 μlrxn**| X samples+1 |	***Final [conc]***  |                                          
|:------------------|:--------------|:-------------|:--------------|
| *Master Mix:*      |                                                                                                 	                                                  
| *5X GC Buffer*    | 	*10.0 μl*|                  | *1 x*|
| *dNTPs (10mM)*	  |   *1 μl*	 |                  | *0.2 mM*|
| *F primer (10uM)* |	*1.25 μl*	 |                  | *0.25 uM*|
| *R primer (10uM)*	|  *1.25 μl* |                  |*0.25 uM*|
 |*H<sub>2</sub>O*|  *32 μl*   |                  |         |
 |**Add last:** 	  |            |                  |         |
 |*HF Phusion Taq*  |	  *0.5μl*  |                  |*1 unit*|
 |*Total*	          | **46.0μl** 	 |                  |           |
 |**Add:**		      |            |                   ||
| *Template DNA*	  |*4.0 μl/well*| 	 ------	      |*50-250 ng*|
 |**TOTAL**	          |**50.0 μl**	 |	                 |          |
 
 |Thermal  |cycling  |conditions|
|:---------:|:---------:|:----------:|
|*98 oC*	  |*3:00 min*	   |           |
|***98 oC***	|***0:10 min***|           |
|***58 oC***	|***0:30 min***	 | ***30X*** |
|***65 oC***	|***5:00 min***	 |       |
|*65 oC*	|*10:00min*	 |           |
|*4 oC* |*∞*|        |   


**Primers:  Pfdhps_F Sequence (5' to 3'):** AATATTTGCGCCAAACTTTTTA; **Pfdhps_R Sequence (5' to 3'):** TTTATTTCGTAATAGTCCACTTTTGAT

***\*Primers adapted from: SC, Carlton JM. 2016.*** A Method for Amplicon Deep Sequencing of Drug Resistance Genes in Plasmodium falciparum Clinical Isolates from India. J Clin Microbiol 54:1500–1511.



## Electrophoresis

This step is necessary to ensure successful amplification of amplicons. It is recommended to run at least 25% of the total samples, all no-template and negative controls on the gel to confirm amplification was successful and no contamination occured. Please note PCR amplification can be affected by numerous factors, including but not limited to, DNA quality and quantity. 

### Consumables
#### **Table7  Electrophoresis Consumables**

|Item                                                   |Quntity                |	Storage|
|:-----------------------------------------------------:|:-----------------------:|:------------------:|                                             
|**Agarose** |1g (for a 1% gel)|Room temperature|
|**1x solution of 10X TBE Buffer and deionized water** | 	100mL (for a 1% gel) | Room temperature|
**Nucleic Acid Gel Stain** |	5 μl per 100mL of buffer |	Room temperature
**Orange Dye** |	2 μl per 8 μl PCR product |	Room temperature

### Preparation
  - Create X solution of 10X TBE Buffer and deionized water before you start.

### Procedure   
1.  Choose an Erlenmeyer flask that is 2-4 times the volume of the solution and place a stirring rod into the flask.
2.  Weigh the agar to the desired concentration
      * For a 1% agarose gel, 1.0 gram of agarose + 100mL of buffer will fill a medium gel chamber
      * For a 1.5% agarose gel, 1.5 gram of agarose + 100mL of buffer will fill a medium gel chamber
3. 	Add the appropriate amount of buffer for the) settled agar.
6. 	Repeat steps 4-5 until all the agar is dissolved (no transparent agarose clumps should be present).
7. 	Allow the solution to cool on a stirring plate until you can comfortably hold the flask with your hands.
8. 	Using a 10 μl pipette, add nucleic acid gel stain to the solution. For every 100mL of buffer, add 5 μl of gel stain. Swirl solution to mix, making sure as little bubbles as possible are created.
9. 	Pour the cooled solution into the gel form- ensure no bubbles are present. Place the comb into the gel and allow the gel to sit undisturbed for at least 15 minutes or until the gel has become firm (the color will change from clear to slightly milky in color).
10. When gel has solidified, ensure the wells are aligned with the black (negative) nodes on the electrophoresis chamber and fill with buffer until it covers about a centimeter above the gel. Remove the comb.
11. Combine a mixture of 2 μl of orange dye and 8 μl of each sample and load 8uL of that mixture into each well.
12. Be sure to include reference ladders (no orange dye necessary).
13. Place the lid on the chamber box and connect the black node to the negative terminal and the red node to the positive terminal. Turn on the power supply and adjust the voltage to 100-130 volts.
14. Run gel for about 45 minutes -until the samples nearly reach the end of the gel.  **DO NOT** allow samples to run off the gel.
15. Turn off the power supply, disconnect the electrodes, and remove the lid.
16. Remove the gel from the chamber and take to the gel reading station for analysis.
17. Once amplification is confirmed, proceed to SequalPrep Normalization **(page 15-16)**.

## SequalPrep Normalization
This step utilizes ChargeSwitch Technology to purify and normalize amplicon PCR product. PCR product is added to the plate and mixed with Binding Buffer, which then sits at room temperature for 1 hour. The DNA is then washed and eluted, resulting in a purified and normalized DNA product. 

This protocol format was adapted from the SequalPrep™ Normalization Plate (96) Kit [protocol](https://www.thermofisher.com/order/catalog/product/A1051001) from ThermoFisher Scientific

### Consumables


#### **Table 8.** PCR Purification 1 SequalPrep Consumables
|                Item                                                 | Quantity                |Storage                          |
|:------------------------------------------------------------------:|:------------------------:|:----------------------------------:|
|**SequalPrep™Normalization Plate (96)**    	                      |1 plate   	         | 15° to 30°C (Room temperature)  |
|**SequalPrep™Normalization Binding Buffer**                          |1 μl per 1 μl of sample   | 15° to 30°C (Room temperature) |
|**SequalPrep™Normalization Wash Buffer** 	                      |0 μl per sample 	         | 15° to 30°C (Room temperature) |
|**SequalPrep™Normalization Elution Buffer(10mM Tris-HCL, pH 8.5)**   |20 μl per sample          |15° to 30°C (Room temperature) |
|**(Optional) Foil Tape** 	                                      |2 sheets                  |15° to 30°C (Room temperature) |

### Preparation

 * You will need at least 250 ng amplicon per well to use with the SequalPrep™ Normalization Plate to achieve robust normalization. An average efficiency PCR (20ul reaction volume) produces in the range of 25-100ng/ul, allowing you to purify using 5-10ul using the SequalPrep system.  

 * Elution options: 
     - The **standard elution** method (described below) is designed to elute purified DNA from each well using 20 μl elution volume to obtain each amplicon at a concentration of 1–2 ng/μl. 
     - The **optional sequential** elution method is designed to sequentially elute multiple rows or columns using the same 20 μl of elution buffer to obtain higher amplicon concentrations. The amplicon concentrations will be additive as sequential wells are eluted. For example, dispense 20 μl of elution buffer into the first column (A1–H1), mix well, and incubate for 5 minutes at room temperature. Then, simply move this column of elution buffer to the next column (A2–H2), and again incubate for 5 minutes. Continue this step to obtain your specific elution needs for the downstream application of choice. 
     

   **NOTE:** Proceed very cautiously during this procedure and take your time- do not introduce bubbles when pipetting. 


### Procedure

#### **Binding Step**

1. 	Transfer the desired volume of PCR product (5–25 μl PCR reaction mix, at least 250 ng amplicon/well) from the PCR plate into the wells of the SequalPrep™ Normalization plate. 
2. 	Add an equivalent volume of SequalPrep™ Normalization Binding Buffer. 
    - For example: To purify 10 μl of PCR product, add 10 μl SequalPrep™ Normalization Binding Buffer.
3. 	Mix completely by pipetting up and down 10 times, or seal the plate with Foil Tape, vortex to mix, and briefly centrifuge the plate.
4. 	Incubate the plate for 1 hour at room temperature to allow binding of DNA to the plate surface. Mixing is not necessary at this stage.
       **NOTE:** Incubations longer than 60 minutes do not improve results. However, depending on your work flow you may perform overnight incubation at room temperature for the binding step.  

5. 	Optional: If >25 ng DNA/well yield is desired, transfer the amplicon/Binding Buffer mixture from Step 4 to another, fresh well/plate to sequentially bind more DNA. Perform DNA binding at room temperature for 1 hour. 
        **NOTE:** After binding is complete, you can remove the amplicon/Binding Buffer mixture from the well and store at –20ºC for up to 30 days to perform additional purifications at a later time.  
        
#### **Washing Step**

6. 	Aspirate the liquid from wells. Be sure not to scrape the well sides during aspiration.
7. 	Add 50 μl SequalPrep™ Normalization Wash Buffer to the wells. Mix by pipetting up and down twice to improve removal of contaminants.
        **NOTE:** If you wish to store the amplicon/Binding Buffer mixture for additional purifications at a later time, aspirate the liquid from wells into another plate and store at –20ºC for up to 30 days.  

8. 	Completely aspirate the buffer from wells and discard. To ensure complete removal of wash buffer and maximize elution efficiency, you may need to invert and tap the plate on paper towels depending on the pipetting technique or instrument used. A small amount of residual Wash Buffer (1–3 μl) is typical and does not affect the subsequent elution or downstream applications.

#### **Elution Step**

9. 	Add 20 μl SequalPrep™ Normalization Elution Buffer to each well of the plate.
       **NOTE:** Do not use water for elution. If you need to elute in any other buffer, be sure to use a buffer of pH 8.5–9.0. If the pH of the buffer is <8.5, the DNA will not elute efficiently.  

10. Mix by pipetting up and down 5 times (or seal the plate with Foil Tape, vortex to mix, and briefly centrifuge the plate). Ensure that the buffer contacts the entire plate coating (up to 20 μl level).
11. Incubate at room temperature for 5 minutes.
12. Transfer the normalized and purified DNA to a new 96-well skirted plate. You can store the eluted DNA at 4°C (short-term storage) or –20°C (long-term storage) until further use. 

13. Pool each drug resistance gene (normalized PCR amplicon) as follows: 
	  - ***10uL of each amplicon (i.e. mitochondria/cyt-b, k13, mdr1, dhfr, dhps, and crt) for each respective sample*** 

Expected DNA concentration is 1–2 ng/μl when using 20 μl elution volume. 



**SAFE STOPPING POINT** If you do not immediately proceed to *Tagment Genomic DNA,* seal plate with Microseal “B” adhesive seal and store it at ‐15° to ‐25°C for up to a week.


## Tagment Genomic DNA
  This step uses the Nextera transposome to tagment gDNA, which fragments and then tags the DNA with adaptor sequences in a single step. 

![alt text](https://github.com/CDCgov/MaRS/blob/master/images/Nextera.png)




### Consumables

#### **Table 9.Tagment Genomic DNA Consumables**

|Item	|Quantity|	Storage|
|:-----------------------------------------:|:--------------------------:|:----------------:|
|**ATM (Amplicon Tagment Mix)**              |	5 μl per sample          |	-15° to -25°C|
|**TD (Tagment DNA Buffer)**	                |10 μl per sample           |	-15° to -25°C|
|**NT (Neutralize Tagment Buffer)**         	| 5 μl per sample       	    |15°C to 30°C|
|**TruSeq Index Plate Fixture (FC-130-1005)**|	1 (if available)            |	Room temperature|
|**96-well 0.2 ml PCR plate**                |	1 plate	                   |Room temperature|
|**Microseal “A” film**                     |	1	                          |Room temperature|


### Preparation
   - If not completed already, carefully pool 10uL of each gene’s corresponding sample into the same well of a half skirt plate before continuing
   - Be sure all samples are mixed thoroughly by pipetting gently 10 times.   
   - Thaw on ice, ATM and TD, invert thawed tubes 3-5 times and then centrifuge briefly. 
   - Check NT for precipitates. If present, vortex until all particulates are re - suspended. 
   - Set up thermal cycler and choose preheat lid option:   ***Thermocycler Program:*** ***55 °C for 5 min***, ***10 °C for ∞***


### Procedure
1. Add the following items in the order listed to each well of a new Hard-Shell skirted PCR plate. Pipette to mix.

|      Item                           |Volume (μl)|
|:------------------------------------:|:----------:|
|   **TD**	                          |   10  |
| **Normalized pooled gene amplicons**|   5   |

2.	Add 5 μl ATM to each well. Pipette to mix.
3.	Centrifuge at 280 × g at 20°C for 1 minute.
4.	Place on the preprogrammed thermal cycler and run the tagmentation program
	***Thermocycler Program:*** ***55 °C for 5 min***, ***10 °C for ∞***
	
  **NOTE:** Be sure to remove the plate as soon as the reaction has completed- NT must be added to the well immediately after the reaction has completed in order to stop the reaction.  

5.	Add 5 μl NT to each well. Pipette to mix.
6. 	Centrifuge at 280 × g at 20°C for 1 minute.
7.  Incubate at room temperature for 5 minutes.
8.	[optional] To assess tagmentation, run 1 uL on Agilent Bioanalyzer and/or Tapestation using high sensitivity DNA chip. 
9.	Proceed to Library Amplification and Index PCR

## Library Amplification and Index PCR
This step amplifies the tagmented DNA using a limited-cycle PCR program. The PCR step adds Index 1 (i7) adapters and Index 2 (i5) adapters and sequences required for cluster formation. Use the full amount of recommended input DNA. To ensure libraries produce high-quality sequencing results, use the specified number of PCR cycles

### Consumables
#### **Table 10. Library Amplification and Index PCR Consumables**

|Item	                                                                                       |Quantity          |	Storage|
|:----------------------------------------------------------------------------------------------|:-----------------|:-----------------|
|**NPM (Nextera PCR Master Mix)**	                                                               |15 μl per sample  |	-15° to -25°C   |
|**Nextera XT Index 1 Primers (N7XX) from the Nextera XT Index kit (FC-131-1001 or FC-131-1002)**|5 μl per sample   |	-15° to -25°C   |
|**Nextera XT Index 2 Primers (S5XX) from the Nextera XT Index kit (FC-131-1001 or FC-131-1002)**|5 μl per sample	  |-15° to -25°C    |
|**96-well 0.2 ml PCR plate**                                                                   |1 plate           |	Room temperature|
|**Microseal “A” film**                                                                        	|1	              |Room temperature|


### Procedure
1. 	Using the entire 25μl from the Tagment Genomic DNA step (see page 17-18), prepare the following: .
2. 	Arrange the Index 1 and 2 primers in a rack (i.e. the TruSeq Index Plate Fixture) using the following arrangements as needed:
    - Arrange Index 2 primer tubes (white caps, clear solution) vertically, aligned with rows A through H.
    - Arrange Index 1 primer tubes (orange caps, yellow solution) horizontally, aligned with columns 1 through 12. For more information on index selection, see Dual Indexing Principle, on page 23.

*\* If no TruSeq Index Plate Fixture is available, arrange the Index adaptors in the same way, and individually pipette each adaptor into its corresponding well*

**Figure 2 *TrueSeq Index Plate Fixture***

![alt text](https://github.com/CDCgov/MaRS/blob/master/images/TruSeqIndex.png)             

         A Index 2 primers (white caps), B Index 1 primers (orange caps), C 96‐well plate.
                




3. 	Place the 96‐well PCR plate with the 25 μl of resuspended PCR product DNA in the TruSeq Index Plate Fixture. 
    -	Add 5 μl of each index into its corresponding well. 

4. 	Add 15uL NPM to each well containing index adaptors. Pipette to mix.  
5.	Centrifuge at 280 × g at 20°C for 1 minute.
6.	Cover the plate with Microseal 'A'.
7.  Perform PCR on a thermal cycler using the following program:
    * 72°C for 3 minutes
    * 95°C for 30 seconds
    * 12 cycles of:
        * 95°C for 10 seconds
        * 55°C for 30 seconds
        * 72°C for 30 seconds
    * 72°C for 5 minutes
    * Hold at 10°C
8.	If you choose to continue, proceed to PCR Clean-up 2




**SAFE STOPPING POINT** If you do not immediately proceed to Library Clean-Up on page 21, seal plate with an adhesive seal and store it at 2° to 8°C for up to a week.

## Library Clean‐Up 
This step uses AMPure XP beads to clean up the final library before quantification.

### Consumables

**Table 11. PCR Purification #2 Consumables**

|Item	                                  |Quantity	                 |Storage                                            |
|:--------------------------------------|:-------------------------|:--------------------------------------------------------|
|**RSB (Resuspension Buffer)**          |	52.5 μl per sample|-15° to -25°C (after initial thaw, can keep at 2° to 8°C)|
|**AMPure XP beads**                    |	90 μl per 50 μl of sample|	2° to 8°C|
|**Freshly Prepared 80% Ethanol (EtOH)**|	400 μl per sample|	Room temperature|
|**96‐well 0.2 ml PCR plate**           |	1 plate                  |                                                         |
|**[Optional] Microseal 'B' film**	    |	                         |                                                         |
|**96‐well MIDI plate**                 |1 plate|	                    |


### Preparation

• Determine whether or not a plate transfer is necessary.  If the PCR reaction volume multiplied by 2.8 exceeds the volume of the PCR plate, a transfer to a 300 μL round bottom plate or a 1.2 mL deep-well plate is required. 

• Thaw RSB at room temperature.  

• Bring the AMPure XP beads to room temperature- wait at least 30 minutes.

• Once at room temperature, shake the Agencourt AMPure XP bottle to re-suspend any magnetic particles that may have settled. Ensure magnetic beads are well (evenly) distributed before adding them to samples. 

**NOTE:** Proceed very cautiously during this procedure and take your time. 

**NOTE:** 70% ethanol is hygroscopic. That is, when opened the ethanol will both evaporate and absorb water over time. Re-use eventually will be at a lower concentration. There is also miscibility involved with ethanol and water. For example, measuring out 70 mL of ethanol and topping off to 100 mL with water will generate ~65% ethanol. Measuring 80 mL ethanol and 20 mL water separately, then combining them will generate ~95 mL of 80% ethanol. Make sure to use molecular biology grade water (DNAse, RNase and Protease free). 

### Procedure
1. 	Centrifuge the Library Amplification plate at 1,000 × g at 20°C for 1 minute to collect condensation, carefully remove seal.
2. 	Using a multichannel pipette, transfer the entire 50 μl of PCR product from the Library Amplification plate to the MIDI plate. Change tips between samples.

**NOTE:** Transfer the sample to a 96‐well MIDI plate if planning to use a shaker for mixing. If mixing by pipette, the sample can remain in the 96‐well PCR plate.  

3. 	Gently shake the AMPure XP beads for 30 seconds to make sure that the beads are evenly dispersed. Add an appropriate volume of beads to a trough depending on the number of samples being processed and desired fragment selection. Smaller amplicons in Nextera XT library preps typically yield smaller insert size ranges. 
To maximize recovery of smaller fragments from the bead cleanup step, use the following conditions: 

![alt text](https://github.com/CDCgov/MaRS/blob/master/images/AMPure.png)

4.  Using a multichannel pipette, add 90 μl of AMPure XP beads to each well of the Amplicon PCR plate. Change tips between columns. 
5.  Gently pipette entire volume up and down 10 times if using a 96‐well PCR plate, or seal plate and shake at 1800 for 2 minutes if using a MIDI plate. Change tips after each column. The mixture should appear homogeneous. 
6.  Incubate the mixed samples at room temperature without shaking for 5 minutes.
7.  Place the Library amplification plate on a magnetic stand for 2 minutes. WAIT for the solution to clear before proceeding.
8.  With the Amplicon PCR plate on the magnetic stand, use a multichannel pipette to carefully remove and discard all the supernatant. Change tips between samples.

	**DO NOT disturb the ring of separated magnetic beads.** 
	
9.  With the Library amplification plate on the magnetic stand, wash the beads with freshly prepared 80% ethanol as follows:
    a. Using a multichannel pipette, add 200 μl of freshly prepared 80% ethanol to each sample well.
    b. Incubate the plate on the magnetic stand for 30 seconds at room temperature.
    c. Carefully remove and discard the ethanol. 
    
     **Note: The beads are not drawn out easily when in alcohol, so it is not necessary to leave any supernatant behind.**

10. With the Library amplification plate on the magnetic stand, perform a second ethanol wash as follows:
    a. Using a multichannel pipette, add 180 μl of freshly prepared 80% ethanol to each sample well.
    b. Incubate the plate on the magnetic stand for 30 seconds.
    c. Carefully remove and discard all the ethanol.
    d. Use a P20 multichannel pipette with fine pipette tips to remove excess ethanol.
11. With the Library amplification plate still on the magnetic stand, allow the beads to air‐dry for 15 minutes.
    
    **NOTE: make sure not to over dry the beads. Bead pellets will appear cracked if over dried.** 
  
12.    Remove the Amplicon PCR plate from the magnetic stand. Using a multichannel pipette, add 52.5 μl RSB to each well of the Amplicon PCR plate.
13.    Gently pipette mix up and down 10 times, or seal plate and shake at 1800 for 2 minutes if using a MIDI plate. Change tips after each column. 
14.    Incubate at room temperature for 2 minutes.
15.    Place the plate back on the magnetic stand for 2 minutes or until the supernatant has cleared.
16.    Using a multichannel pipette, carefully transfer 50 μl of the supernatant from the Library amplification plate to a new 96‐well PCR plate. Change tips between samples to avoid cross‐contamination.


**SAFE STOPPING POINT**
If you do not plan to proceed to Library Clustering on page 24, seal the plate with Microseal “B” adhesive seal. Store the plate at ‐15° to ‐25°C for up to a week.














                                                             
                                                            
                                                            
                                                            











































